### PR TITLE
Convert additional templates to Jinja

### PR DESCRIPTION
**Pull request by clokep**
This converts a few more of our inline HTML templates to Jinja. This is somewhat part of #7280 and should make it a bit easier to customize these in the future.

### CODE REVIEW

File: synapse/config/_base.py [hunk 1]
```diff
@@ -242,12 +242,11 @@ def read_templates(
...
-            {
-                "format_ts": _format_ts_filter,
-                "mxc_to_http": _create_mxc_to_http_filter(self.public_baseurl),
-            }
-        )
+        env.filters.update({"format_ts": _format_ts_filter})
+        if self.public_baseurl:
+            env.filters.update(
+                {"mxc_to_http": _create_mxc_to_http_filter(self.public_baseurl)}
+            )
```
Comment (clokep) [inline synapse/config/_base.py:249]: Per the conversation in #synapse-dev:matrix.org it seems that `public_baseurl` is not required to be set. I believe any feature that expects to use this template filter already asserts that.

Comment (anoadragon453) [reply synapse/config/_base.py:249]: Seems so!

**Review by anoadragon453 — APPROVED** [review]

**Comment by michaelkaye** [comment]
I don't know if it's worth renaming the templates to indicate that they're not pure HTML - e.g. ansible community seems to have gravitated towards  ".html.j2" for something that needs templating before it's usable.

**Comment by anoadragon453** [comment]
@michaelkaye We have a lot of templates that would need to be renamed if that were the case, and it might be a bit strange having only these 3 as `html.j2` and the rest as `html`. Perhaps we can do so in a separate PR where we search for and support both extensions?

**Comment by clokep** [comment]
> I don't know if it's worth renaming the templates to indicate that they're not pure HTML - e.g. ansible community seems to have gravitated towards ".html.j2" for something that needs templating before it's usable.

I agree that it is confusing, but it has generally been our convention to just name them ".html". I added a note to #7280 though!.

**Pull request merged**